extensions/amp-analytics/0.1/base64.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {base64ToUrlSafe} from '../../../src/utils/base64';
+
+/**
+ * Base64 encoding that can handle unicode strings in a byte-efficent manner. As
+ * recommended by MDN;
+ * https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding
+ * @param {str} str
+ * @return {str}
+ */
+export function b64EncodeUnicode(str) {
```

Comment:
This is basically `base64EncodeFromBytes`.

I think you having written this is really caused by how bad the APIs provided by `src/utils/base64.js` are (if you can't tell, I hate every one of the helpers in there). None of them should be concerned with "bytes" (`Uint8Array`), it should just be string input, string output. If you cleaned that up, I'd be super happy.

Reply:
I actually just realized the same thing :). I think a function that wraps `utf8Encode` and `base64UrlEncodeFromBytes` is the way to go.